pkg/virtualKubelet/provider/pods.go

```diff
@@ -23,122 +24,119 @@ import (
 )
 
 // CreatePod accepts a Pod definition and stores it in memory.
-func (p *KubernetesProvider) CreatePod(ctx context.Context, pod *v1.Pod) error {
-	// Add the pod's coordinates to the current span.
-	if pod == nil {
-		return errors.New("pod cannot be nil")
+func (p *LiqoProvider) CreatePod(_ context.Context, homePod *corev1.Pod) error {
+	if homePod == nil {
+		return errors.New("received nil homePod to create")
 	}
 
-	klog.Infof("receive CreatePod %q", pod.Name)
+	klog.V(3).Infof("PROVIDER: pod %s/%s asked to be created in the provider", homePod.Namespace, homePod.Name)
 
-	if pod.OwnerReferences != nil && len(pod.OwnerReferences) != 0 && pod.OwnerReferences[0].Kind == "DaemonSet" {
-		msg := fmt.Sprintf("Skip to create DaemonSet pod %q", pod.Name)
-		klog.Info(msg)
+	if homePod.OwnerReferences != nil && len(homePod.OwnerReferences) != 0 && homePod.OwnerReferences[0].Kind == "DaemonSet" {
+		klog.Infof("Skip to create DaemonSet homePod %q", homePod.Name)
 		return nil
 	}
 
-	nattedNS, err := p.namespaceMapper.NatNamespace(pod.Namespace, true)
+	foreignPod, err := forge.HomeToForeign(homePod, nil, forge.LiqoOutgoing)
 	if err != nil {
 		return err
 	}
 
-	podTranslated := translation.H2FTranslate(pod, nattedNS)
+	foreignReplicaset := forge.ReplicasetFromPod(foreignPod.(*corev1.Pod))
 
-	apiController, err := p.GetApiController()
-	if err != nil {
-		return err
-	}
-	podTranslated, err = serviceEnv.TranslateServiceEnvVariables(podTranslated, pod.Namespace, nattedNS, apiController.CacheManager())
+	_, err = p.foreignClient.AppsV1().ReplicaSets(foreignReplicaset.Namespace).Create(context.TODO(), foreignReplicaset, metav1.CreateOptions{})
```

Comment:
Probably we are missing service environment translations

Reply:
Nice catch, we were going to miss them.  I introduced them again. Anyway, I have some doubts about that function: what happens if, let's say one over ten services hasn't been replicated yet? The translated pod will have all the service environment variables but one?

Reply:
Yes, good point, but I think that is the same for local deployment. If the pod comes before that a service is available it can't have it in env var but has to discover it with DNS

I think that in all the cases you have to be sure that the service was available a lot of time before the creation of the pod

Reply:
Yes, I suppose there is the same issue even in local deployments.